### ISSUE TITLE
[FritzboxTr064]: Error while executing background thread FritzboxTr064 Refresh Service

### ISSUE THREAD

**Issue by staehler** [openhab/openhab2-addons#2784]
After some time running openhab this error appears:
```
2017-10-16 11:21:12.694 [ERROR] [b.core.service.AbstractActiveService] - Error while executing background thread FritzboxTr064 Refresh Service
java.lang.IllegalStateException: Auth scheme is null
        at org.apache.http.util.Asserts.notNull(Asserts.java:52) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.auth.HttpAuthenticator.ensureAuthScheme(HttpAuthenticator.java:229) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.auth.HttpAuthenticator.generateAuthResponse(HttpAuthenticator.java:184) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:262) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.openhab.binding.fritzboxtr064.internal.Tr064Comm.readSoapResponse(Tr064Comm.java:456) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.openhab.binding.fritzboxtr064.internal.Tr064Comm.getTr064Values(Tr064Comm.java:247) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.openhab.binding.fritzboxtr064.internal.FritzboxTr064Binding.execute(FritzboxTr064Binding.java:197) [209:org.openhab.binding.fritzboxtr064:1.11.0.201710050110]
        at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:144) [178:org.openhab.core.compat1x:2.2.0.201710112009]
        at org.openhab.core.service.AbstractActiveService$RefreshThread.run(AbstractActiveService.java:166) [178:org.openhab.core.compat1x:2.2.0.201710112009]
```

## Expected Behavior
This error is new, as it worked for month before the upgrade. I made an upgrade from openhab2_2.2.0-20170725225428-1_all.deb to openhab2_2.2.0-20171011224825-1. So this behaviour must have been imported in between July and October.

## Current Behavior
When this error happens, it doesn't disappear untill a restart of openhab or of the tr064 bundle in karaf.
With this error, no actions work (like switch WLAN ON/OFF).

## Workaround
Restart openhab or restart tr064 bundle

## Steps to Reproduce (for bugs)
I can not reproduce it. It seems to be dependent on the time running this binding.

## Your Environment
Debian GNU/Linux 8
openhab2_2.2.0-20171011224825-1
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)

**Comment by jschanz** [comment]
Problem exists also in openHAB-2.1. A "bundle:restart 200" will solve this problem for now, until it occures a few hours / days later again.

**Comment by jschanz** [comment]
Solved this on a 7390 by falling back to http and a refresh interval of 60.000 ms

```# URL. Either use http://<fbox-ip>:49000 or https://<fbox-ip>:49443 (https preferred!)
# url=https://192.168.178.1:49443
url=http://192.168.178.1:49000

# Refresh Interval (60000ms default)
refresh=60000
```

**Comment by oprobst** [comment]
I observed the same behavior on my installation. In my case, I think I found the problem: Looks like it's located at Tr064Comm.java line ):
`
   // Check for (auth-) error
            if (slResponse.getStatusCode() == 401) {
                logger.error(
                        "Could not read response from FritzBox. Unauthorized! Check User/PW in config. Create user for tr064 requests");
                _httpClientContext.getTargetAuthState().reset();
                return null;
            }
`

This is exactly the last error message I got in log before the Auth Scheme is invalid. I assume that `_httpClientContext.getTargetAuthState().reset();` invalidates the authorization, but do not clean up the connection itself.

I'm going to verify this for a few days and will create a Pull Request in case it solves the problem.
